File: design-system/android/components/DSButton.kt
package com.designsystem.components

import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.scale
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.painter.Painter
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.TextUnit
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.designsystem.theme.DSColors
import com.designsystem.theme.DSTypography

// Button Variants
enum class DSButtonVariant {
    PRIMARY,
    SECONDARY,
    TERTIARY,
    GHOST,
    DANGER
}

// Button Sizes
enum class DSButtonSize {
    SMALL,
    MEDIUM,
    LARGE
}

// Icon Position
enum class IconPosition {
    LEADING,
    TRAILING
}

// Button Size Properties
data class ButtonSizeProperties(
    val height: Dp,
    val fontSize: TextUnit,
    val horizontalPadding: Dp,
    val iconSize: Dp,
    val minWidth: Dp
)

// Get size properties for button
private fun getButtonSizeProperties(size: DSButtonSize): ButtonSizeProperties {
    return when (size) {
        DSButtonSize.SMALL -> ButtonSizeProperties(
            height = 32.dp,
            fontSize = 14.sp,
            horizontalPadding = 12.dp,
            iconSize = 16.dp,
            minWidth = 64.dp
        )
        DSButtonSize.MEDIUM -> ButtonSizeProperties(
            height = 40.dp,
            fontSize = 16.sp,
            horizontalPadding = 16.dp,
            iconSize = 20.dp,
            minWidth = 80.dp
        )
        DSButtonSize.LARGE -> ButtonSizeProperties(
            height = 48.dp,
            fontSize = 18.sp,
            horizontalPadding = 24.dp,
            iconSize = 24.dp,
            minWidth = 96.dp
        )
    }
}

// Get colors for button variant
private fun getButtonColors(variant: DSButtonVariant): ButtonColors {
    return when (variant) {
        DSButtonVariant.PRIMARY -> ButtonDefaults.buttonColors(
            containerColor = DSColors.Primary500,
            contentColor = Color.White,
            disabledContainerColor = DSColors.Primary500.copy(alpha = 0.5f),
            disabledContentColor = Color.White.copy(alpha = 0.5f)
        )
        DSButtonVariant.SECONDARY -> ButtonDefaults.buttonColors(
            containerColor = DSColors.Secondary500,
            contentColor = Color.White,
            disabledContainerColor = DSColors.Secondary500.copy(alpha = 0.5f),
            disabledContentColor = Color.White.copy(alpha = 0.5f)
        )
        DSButtonVariant.TERTIARY -> ButtonDefaults.outlinedButtonColors(
            containerColor = Color.Transparent,
            contentColor = DSColors.Primary600,
            disabledContentColor = DSColors.Primary600.copy(alpha = 0.5f)
        )
        DSButtonVariant.GHOST -> ButtonDefaults.textButtonColors(
            containerColor = Color.Transparent,
            contentColor = DSColors.TextPrimary,
            disabledContentColor = DSColors.TextPrimary.copy(alpha = 0.5f)
        )
        DSButtonVariant.DANGER -> ButtonDefaults.buttonColors(
            containerColor = DSColors.ErrorMain,
            contentColor = Color.White,
            disabledContainerColor = DSColors.ErrorMain.copy(alpha = 0.5f),
            disabledContentColor = Color.White.copy(alpha = 0.5f)
        )
    }
}

// Main Button Composable
@Composable
fun DSButton(
    text: String? = null,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    variant: DSButtonVariant = DSButtonVariant.PRIMARY,
    size: DSButtonSize = DSButtonSize.MEDIUM,
    enabled: Boolean = true,
    loading: Boolean = false,
    fullWidth: Boolean = false,
    icon: Painter? = null,
    iconPosition: IconPosition = IconPosition.LEADING
) {
    val sizeProperties = getButtonSizeProperties(size)
    val colors = getButtonColors(variant)
    var isPressed by remember { mutableStateOf(false) }
    val scale by animateFloatAsState(if (isPressed) 0.98f else 1f)
    
    val buttonModifier = modifier
        .scale(scale)
        .then(
            if (fullWidth) Modifier.fillMaxWidth()
            else Modifier.defaultMinSize(minWidth = sizeProperties.minWidth)
        )
        .height(sizeProperties.height)
    
    val border = if (variant == DSButtonVariant.TERTIARY) {
        BorderStroke(1.dp, DSColors.Primary600)
    } else null
    
    Button(
        onClick = {
            if (!loading && enabled) {
                onClick()
            }
        },
        modifier = buttonModifier,
        enabled = enabled && !loading,
        colors = colors,
        shape = RoundedCornerShape(6.dp),
        border = border,
        contentPadding = PaddingValues(horizontal = sizeProperties.horizontalPadding)
    ) {
        if (loading) {
            CircularProgressIndicator(
                modifier = Modifier.size(sizeProperties.iconSize),
                color = LocalContentColor.current,
                strokeWidth = 2.dp
            )
        } else {
            Row(
                horizontalArrangement = Arrangement.Center,
                verticalAlignment = Alignment.CenterVertically
            ) {
                if (icon != null && iconPosition == IconPosition.LEADING) {
                    Icon(
                        painter = icon,
                        contentDescription = null,
                        modifier = Modifier.size(sizeProperties.iconSize)
                    )
                    if (text != null) {
                        Spacer(modifier = Modifier.width(8.dp))
                    }
                }
                
                if (text != null) {
                    Text(
                        text = text,
                        fontSize = sizeProperties.fontSize,
                        fontWeight = FontWeight.Medium,
                        maxLines = 1
                    )
                }
                
                if (icon != null && iconPosition == IconPosition.TRAILING) {
                    if (text != null) {
                        Spacer(modifier = Modifier.width(8.dp))
                    }
                    Icon(
                        painter = icon,
                        contentDescription = null,
                        modifier = Modifier.size(sizeProperties.iconSize)
                    )
                }
            }
        }
    }
}

// Button Group Composable
@Composable
fun DSButtonGroup(
    modifier: Modifier = Modifier,
    orientation: Orientation = Orientation.HORIZONTAL,
    spacing: Dp = 8.dp,
    content: @Composable () -> Unit
) {
    when (orientation) {
        Orientation.HORIZONTAL -> {
            Row(
                modifier = modifier,
                horizontalArrangement = Arrangement.spacedBy(spacing)
            ) {
                content()
            }
        }
        Orientation.VERTICAL -> {
            Column(
                modifier = modifier,
                verticalArrangement = Arrangement.spacedBy(spacing)
            ) {
                content()
            }
        }
    }
}

enum class Orientation {
    HORIZONTAL,
    VERTICAL
}

// Icon-only Button Composable
@Composable
fun DSIconButton(
    icon: Painter,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    variant: DSButtonVariant = DSButtonVariant.PRIMARY,
    size: DSButtonSize = DSButtonSize.MEDIUM,
    enabled: Boolean = true,
    contentDescription: String? = null
) {
    val sizeProperties = getButtonSizeProperties(size)
    val colors = getButtonColors(variant)
    
    val buttonSize = when (size) {
        DSButtonSize.SMALL -> 32.dp
        DSButtonSize.MEDIUM -> 40.dp
        DSButtonSize.LARGE -> 48.dp
    }
    
    if (variant == DSButtonVariant.GHOST) {
        IconButton(
            onClick = onClick,
            modifier = modifier.size(buttonSize),
            enabled = enabled
        ) {
            Icon(
                painter = icon,
                contentDescription = contentDescription,
                modifier = Modifier.size(sizeProperties.iconSize),
                tint = colors.contentColor
            )
        }
    } else {
        FilledIconButton(
            onClick = onClick,
            modifier = modifier.size(buttonSize),
            enabled = enabled,
            colors = IconButtonDefaults.filledIconButtonColors(
                containerColor = colors.containerColor,
                contentColor = colors.contentColor,
                disabledContainerColor = colors.disabledContainerColor,
                disabledContentColor = colors.disabledContentColor
            ),
            shape = RoundedCornerShape(6.dp)
        ) {
            Icon(
                painter = icon,
                contentDescription = contentDescription,
                modifier = Modifier.size(sizeProperties.iconSize)
            )
        }
    }
}

File: design-system/ios/components/DSButton.swift
import SwiftUI

// MARK: - Button Style
public enum DSButtonVariant {
    case primary
    case secondary
    case tertiary
    case ghost
    case danger
}

public enum DSButtonSize {
    case small
    case medium
    case large
    
    var height: CGFloat {
        switch self {
        case .small: return 32
        case .medium: return 40
        case .large: return 48
        }
    }
    
    var fontSize: Font {
        switch self {
        case .small: return .system(size: 14, weight: .medium)
        case .medium: return .system(size: 16, weight: .medium)
        case .large: return .system(size: 18, weight: .medium)
        }
    }
    
    var horizontalPadding: CGFloat {
        switch self {
        case .small: return 12
        case .medium: return 16
        case .large: return 24
        }
    }
    
    var iconSize: CGFloat {
        switch self {
        case .small: return 16
        case .medium: return 20
        case .large: return 24
        }
    }
}

// MARK: - Button Component
public struct DSButton: View {
    let title: String?
    let icon: Image?
    let iconPosition: IconPosition
    let variant: DSButtonVariant
    let size: DSButtonSize
    let isFullWidth: Bool
    let isLoading: Bool
    let isDisabled: Bool
    let action: () -> Void
    
    public enum IconPosition {
        case leading
        case trailing
    }
    
    public init(
        title: String? = nil,
        icon: Image? = nil,
        iconPosition: IconPosition = .leading,
        variant: DSButtonVariant = .primary,
        size: DSButtonSize = .medium,
        isFullWidth: Bool = false,
        isLoading: Bool = false,
        isDisabled: Bool = false,
        action: @escaping () -> Void
    ) {
        self.title = title
        self.icon = icon
        self.iconPosition = iconPosition
        self.variant = variant
        self.size = size
        self.isFullWidth = isFullWidth
        self.isLoading = isLoading
        self.isDisabled = isDisabled
        self.action = action
    }
    
    public var body: some View {
        Button(action: {
            if !isDisabled && !isLoading {
                action()
            }
        }) {
            HStack(spacing: 8) {
                if isLoading {
                    ProgressView()
                        .progressViewStyle(CircularProgressViewStyle(tint: foregroundColor))
                        .scaleEffect(0.8)
                } else {
                    if iconPosition == .leading, let icon = icon {
                        icon
                            .resizable()
                            .aspectRatio(contentMode: .fit)
                            .frame(width: size.iconSize, height: size.iconSize)
                    }
                    
                    if let title = title {
                        Text(title)
                            .font(size.fontSize)
                            .lineLimit(1)
                    }
                    
                    if iconPosition == .trailing, let icon = icon {
                        icon
                            .resizable()
                            .aspectRatio(contentMode: .fit)
                            .frame(width: size.iconSize, height: size.iconSize)
                    }
                }
            }
            .frame(maxWidth: isFullWidth ? .infinity : nil)
            .frame(height: size.height)
            .padding(.horizontal, size.horizontalPadding)
            .foregroundColor(foregroundColor)
            .background(backgroundColor)
            .cornerRadius(6)
            .overlay(
                RoundedRectangle(cornerRadius: 6)
                    .stroke(borderColor, lineWidth: hasBorder ? 1 : 0)
            )
        }
        .disabled(isDisabled || isLoading)
        .opacity(isDisabled ? 0.5 : 1.0)
        .scaleEffect(isPressed ? 0.98 : 1.0)
        .animation(.easeInOut(duration: 0.1), value: isPressed)
    }
    
    // MARK: - State
    @State private var isPressed: Bool = false
    
    // MARK: - Colors
    private var backgroundColor: Color {
        switch variant {
        case .primary:
            return DSColors.primary500
        case .secondary:
            return DSColors.secondary500
        case .tertiary:
            return Color.clear
        case .ghost:
            return Color.clear
        case .danger:
            return DSColors.errorMain
        }
    }
    
    private var foregroundColor: Color {
        switch variant {
        case .primary, .secondary, .danger:
            return .white
        case .tertiary:
            return DSColors.primary600
        case .ghost:
            return DSColors.textPrimary
        }
    }
    
    private var borderColor: Color {
        switch variant {
        case .tertiary:
            return DSColors.primary600
        default:
            return Color.clear
        }
    }
    
    private var hasBorder: Bool {
        variant == .tertiary
    }
}

// MARK: - Button Group
public struct DSButtonGroup<Content: View>: View {
    public enum Orientation {
        case horizontal
        case vertical
    }
    
    public enum Spacing {
        case none
        case small
        case medium
        case large
        
        var value: CGFloat {
            switch self {
            case .none: return 0
            case .small: return 8
            case .medium: return 16
            case .large: return 24
            }
        }
    }
    
    let orientation: Orientation
    let spacing: Spacing
    let content: Content
    
    public init(
        orientation: Orientation = .horizontal,
        spacing: Spacing = .small,
        @ViewBuilder content: () -> Content
    ) {
        self.orientation = orientation
        self.spacing = spacing
        self.content = content()
    }
    
    public var body: some View {
        Group {
            if orientation == .horizontal {
                HStack(spacing: spacing.value) {
                    content
                }
            } else {
                VStack(spacing: spacing.value) {
                    content
                }
            }
        }
    }
}

// MARK: - Preview
struct DSButton_Previews: PreviewProvider {
    static var previews: some View {
        VStack(spacing: 20) {
            DSButton(title: "Primary Button", variant: .primary) {}
            DSButton(title: "Secondary Button", variant: .secondary) {}
            DSButton(title: "Tertiary Button", variant: .tertiary) {}
            DSButton(title: "Ghost Button", variant: .ghost) {}
            DSButton(title: "Danger Button", variant: .danger) {}
            
            DSButton(
                title: "Button with Icon",
                icon: Image(systemName: "star.fill"),
                variant: .primary
            ) {}
            
            DSButton(
                title: "Loading Button",
                variant: .primary,
                isLoading: true
            ) {}
            
            DSButton(
                title: "Disabled Button",
                variant: .primary,
                isDisabled: true
            ) {}
            
            DSButton(
                title: "Full Width Button",
                variant: .primary,
                isFullWidth: true
            ) {}
            
            DSButtonGroup(orientation: .horizontal) {
                DSButton(title: "Cancel", variant: .ghost) {}
                DSButton(title: "Save", variant: .primary) {}
            }
        }
        .padding()
    }
}